### DISCUSSION TITLE
Newbie fabric: true question

### DISCUSSION THREAD

**Discussion by schriker** [mateusz1913/rnbridgingtutorial#50] [Q&A]
HI, 👋 First I would say thank you for your work, great tutorial! I'm putting this here because it doesn't feel like an issue but I would ask some questions. Never worked before with new architecture and I have some questions.

I'm setting up `AppInfo` module and everything is working fine but after running `npx pod-install` I no longer see `"fabric":true` from metro after running app. 

If i try to run again `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` to enable new architecture I'm getting this error: 

```The Swift pod `AppInfoPackage` depends upon `React-Fabric`, `React-graphics`, `React-utils`, and `React-debug`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.```

This is happening if i choose `ObjectiveC + Swift` if I go only with `ObjectiveC` setup I'm able to install my pods with `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` and can see `"fabric":true` without any issues.

So my questions:

1. Is it expected to NOT see `"fabric":true` after `npx pod-install`?
2. Let's say I would like to install some 3rd party library later, should I just run `pod install` from iOS folder or after enabling new architecture I always have to run pos install like this: `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`

**Comment by mateusz1913** [comment]
Hey @schriker which version of the RN do you use while getting the "modular headers" error? I remember that there were a few commits added to RN core repo to prevent such issues

**Reply by schriker** [reply]
Hi, thanks for response 🙏 I have created repo: https://github.com/schriker/bridging this is just for iOS and right before this step: https://mateusz1913.github.io/rnbridgingtutorial/docs/guides/app-info-module/setup#ios-pods-installation if I run `RCT_NEW_ARCH_ENABLED=1 npx pod-install` then I got this error: 

<img width="733" alt="image" src="https://github.com/mateusz1913/rnbridgingtutorial/assets/28902144/ea706c8b-2ffc-4fae-b41e-e24027f3ffd0">

This is `"react-native": "0.72.3"`, again this is happening only if I'm doing ObjC+Swift setup. `npx pod-install` is working fine but then I no longer use new arch, if I'm understanding correctly.

**Reply by mateusz1913** [reply]
I found the PR that was merged a while ago to RN core, however it was reverted after some time - https://github.com/facebook/react-native/pull/38121

You can check it for more context. Tbh, I don't have an idea how to solve it

**Comment by mateusz1913** [comment]
Ad 1. The `{fabric: true}` on metro console is only expected to happen when the `RCT_NEW_ARCH_ENABLED` is set during pod installation

Ad 2. you can do it either with command execution or put it inside Podfile (`ENV["RCT_NEW_ARCH_ENABLED"] = "1"`) like it's described in the tutorial's getting started section (so that you don't need to type it before each `npx pod-install` execution)
